CLI/controllers/sendSchemaController.go

```diff
@@ -20,106 +20,86 @@ func serialiseVector(attributes map[string]interface{}, attributeName string) {
 	}
 }
 
-// Serialising size & posXY is inefficient but
-// the team wants it for now
-// "size":"[25,29.4,0]" -> "size": "{\"x\":25,\"y\":29.4,\"z\":0}"
 func serialiseStringVector(attr map[string]interface{}, want string) string {
 	var newSize string
-	if size, ok := attr[want].(string); ok {
-		left := strings.Index(size, "[")
-		right := strings.Index(size, "]")
-		coords := []string{"x", "y", "z"}
 
-		if left != -1 && right != -1 {
-			var length int
-			subStr := size[left+1 : right]
-			nums := stringSplitter(subStr, ",", want)
-			if nums == nil { //Error!
-				return ""
-			}
-			//nums := strings.Split(subStr, ",")
+	size, ok := attr[want].(string)
+	if !ok {
+		return ""
+	}
 
-			if len(nums) == 3 && want == "size" {
-				length = 2
-			} else {
-				//Make posXYZ length 3 for racks
-				if want == "posXYZ" && len(nums) == 2 {
-					nums = append(nums, "0")
-				}
-				length = len(nums)
-			}
+	left := strings.Index(size, "[")
+	right := strings.Index(size, "]")
+	if left == -1 || right == -1 {
+		return ""
+	}
 
-			for idx := 0; idx < length; idx++ {
-				newSize += "\"" + coords[idx] + "\":" + nums[idx]
+	nums := stringSplitter(size[left+1:right], ",", want)
+	if nums == nil {
+		return ""
+	}
+	//nums := strings.Split(subStr, ",")
 
-				if idx < length-1 {
-					newSize += ","
-				}
-			}
-			newSize = "{" + newSize + "}"
+	length := len(nums)
+	if length == 3 && want == "size" {
+		length = 2
+	} else if want == "posXYZ" && length == 2 {
+		nums = append(nums, "0")
+		length++
+	}
 
-			if len(nums) == 3 && want == "size" {
-				if attr["shape"] == "sphere" || attr["shape"] == "cylinder" {
-					attr["diameter"] = nums[2]
-					if attr["shape"] == "cylinder" {
-						attr["height"] = nums[1]
-					}
-					return ""
-				} else {
-					attr["height"] = nums[2]
-				}
+	newSize = "[" + strings.Join(nums[:length], ", ") + "]"
+
+	if len(nums) == 3 && want == "size" {
+		if attr["shape"] == "sphere" || attr["shape"] == "cylinder" {
+			attr["diameter"] = nums[2]
+			if attr["shape"] == "cylinder" {
+				attr["height"] = nums[1]
 			}
+			return ""
+		} else {
+			attr["height"] = nums[2]
 		}
 	}
+
 	return newSize
 }
 
-// Same utility func as above but we have an arbitrary array
-// and want to cast it to -> "size": "{\"x\":25,\"y\":29.4,\"z\":0}"
 func serialiseFloatVector(attr map[string]interface{}, want string) string {
 	var newSize string
-	if items, ok := attr[want].([]float64); ok {
-		coords := []string{"x", "y", "z"}
-		var length int
 
-		if isValid := arrayVerifier(&items, want); !isValid {
-			return ""
-		}
+	items, ok := attr[want].([]float64)
+	if !ok || !arrayVerifier(&items, want) {
+		return ""
+	}
 
-		if len(items) == 3 && want == "size" {
-			length = 2
-		} else {
-			//Make posXYZ length 3 for racks
-			if want == "posXYZ" && len(items) == 2 {
-				items = append(items, 0)
-			}
-			length = len(items)
-		}
+	length := len(items)
+	if length == 3 && want == "size" {
```

Comment:
Basically repeats the code from above, are we sure both are used? do we need both?

Reply:
Both are indeed used, but your review gave me an idea and I've managed to unify both functions thanks to the handy `Stringify` function in the same file. :)